control/phaseplot.py

```diff
@@ -1,61 +1,912 @@
-#! TODO: add module docstring
 # phaseplot.py - generate 2D phase portraits
 #
 # Author: Richard M. Murray
-# Date: 24 July 2011, converted from MATLAB version (2002); based on
-# a version by Kristi Morgansen
-#
-# Copyright (c) 2011 by California Institute of Technology
-# All rights reserved.
-#
-# Redistribution and use in source and binary forms, with or without
-# modification, are permitted provided that the following conditions are
-# met:
+# Date:   23 Mar 2024 (legacy version information below)
 #
-#   1. Redistributions of source code must retain the above copyright
-#      notice, this list of conditions and the following disclaimer.
-#
-#   2. Redistributions in binary form must reproduce the above copyright
-#      notice, this list of conditions and the following disclaimer in the
-#      documentation and/or other materials provided with the distribution.
-#
-#   3. The name of the author may not be used to endorse or promote products
-#      derived from this software without specific prior written permission.
-#
-# THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
-# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
-# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
-# DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
-# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
-# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
-# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
-# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
-# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
-# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
-# POSSIBILITY OF SUCH DAMAGE.
+# TODO
+# * Allow multiple timepoints (and change timespec name to T?)
+# * Update linestyles (color -> linestyle?)
+# * Check for keyword compatibility with other plot routines
+# * Set up configuration parameters (nyquist --> phaseplot)
 
-import numpy as np
-import matplotlib.pyplot as mpl
+"""The :mod:`control.phaseplot` module contains functions for generating 2D
+phase plots.
```

Comment:
Convention is for summary to be 1 line (<https://peps.python.org/pep-0257/#multi-line-docstrings>).